libraries/AP_Math/SCurve.cpp

```diff
@@ -52,23 +52,66 @@ void SCurve::init()
     time = 0.0f;
     num_segs = SEG_INIT;
     add_segment(num_segs, 0.0f, SegmentType::CONSTANT_JERK, 0.0f, 0.0f, 0.0f, 0.0f);
-    track.zero();
-    delta_unit.zero();
-    position_sq = 0.0f;
+
+    is_arc_segment = false;
+    seg_delta.zero();
+    seg_length = 0.0;
+    arc.angle_rad = 0.0;
+    arc.length_ne = 0.0;
```

Comment:
any reason to not zero radius_ne as well?

Reply:
I think you could actually shorten it to what's below to zero everything in the structure

arc = {};